woocommerce/woocommerce
# Add menu item to Publish button with copy to draft

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds the `Copy to a new draft` option to the selector next to the `Publish` button.

![Screenshot 2024-04-08 at 10 52 17 AM](https://github.com/woocommerce/woocommerce/assets/1314156/8708af3b-eab3-4d6e-b6b2-3f6ea9c765f3)

Closes #43062.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the New product editor is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Product > Add New and create different types of products (simple, variable, etc).
3. Go to the products and verify that the `Copy to a new draft` option works well for every product type.
4. Go to Product > Add New.
5. Add a product name and press `Copy to a new draft`.
6. The product name should be copied to the new product with the `(copy)` at the end. That only happens with the name of new products.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
